src/buffer/buffer_stm_test.ml

```diff
@@ -0,0 +1,123 @@
+open QCheck
+open Util
+
+(** parallel STM tests of Buffer *)
+
+(* port from the QCSTM example *)
+module BConf =
+struct
+  type cmd =
+    | Contents
+    (* To_bytes | Sub | Blit *)
+    | Nth of int
+    | Length
+    | Clear
+    | Reset
+    | Add_char of char
+    (* Add_utf8_uchar | Add_utf_16le_uchar | Add_utf_16be_uchar *)
+    | Add_string of string
+    (*| Add_bytes of bytes*)
```

Comment:
I know this is a port from `qcstm` - so this is not to point any fingers.
Still: how hard would it be to include the remaining bindings from `Buffer`'s signature?